### ISSUE TITLE
Strange exception when connecting to modem (sometimes)

### ISSUE THREAD

**Issue by GideonLeGrange** [GideonLeGrange/panstamp-java#9]
```
me.legrange.panstamp.ModemException: Malformed integer response '' to ATCH? comamnd
	at me.legrange.panstamp.Network.getSetup(Network.java:459)
	at me.legrange.panstamp.Network.open(Network.java:96)
	at me.legrange.panstamp.Network.openSerial(Network.java:45)
	at me.legrange.panstamp.gui.NetworkAddDialog.createNetwork(NetworkAddDialog.java:620)
	at me.legrange.panstamp.gui.NetworkAddDialog.finishButtonActionPerformed(NetworkAddDialog.java:494)
	at me.legrange.panstamp.gui.NetworkAddDialog.access$1400(NetworkAddDialog.java:27)
	at me.legrange.panstamp.gui.NetworkAddDialog$15.actionPerformed(NetworkAddDialog.java:440)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2346)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)
	at java.awt.Component.processMouseEvent(Component.java:6525)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3321)
	at java.awt.Component.processEvent(Component.java:6290)
	at java.awt.Container.processEvent(Container.java:2234)
	at java.awt.Component.dispatchEventImpl(Component.java:4881)
	at java.awt.Container.dispatchEventImpl(Container.java:2292)
	at java.awt.Component.dispatchEvent(Component.java:4703)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4898)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4533)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4462)
	at java.awt.Container.dispatchEventImpl(Container.java:2278)
	at java.awt.Window.dispatchEventImpl(Window.java:2739)
	at java.awt.Component.dispatchEvent(Component.java:4703)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:746)
	at java.awt.EventQueue.access$400(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:697)
	at java.awt.EventQueue$3.run(EventQueue.java:691)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:719)
	at java.awt.EventQueue$4.run(EventQueue.java:717)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:716)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:109)
	at java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:184)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:229)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:227)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:227)
	at java.awt.Dialog.show(Dialog.java:1084)
	at java.awt.Component.show(Component.java:1654)
	at java.awt.Component.setVisible(Component.java:1606)
	at java.awt.Window.setVisible(Window.java:1014)
	at java.awt.Dialog.setVisible(Dialog.java:1005)
	at me.legrange.panstamp.gui.view.View.showNetworkAddDialog(View.java:27)
	at me.legrange.panstamp.gui.view.Menus$1.actionPerformed(Menus.java:150)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2346)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.AbstractButton.doClick(AbstractButton.java:376)
	at javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:833)
	at com.apple.laf.AquaMenuItemUI.doClick(AquaMenuItemUI.java:157)
	at javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:877)
	at java.awt.Component.processMouseEvent(Component.java:6525)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3321)
	at java.awt.Component.processEvent(Component.java:6290)
	at java.awt.Container.processEvent(Container.java:2234)
	at java.awt.Component.dispatchEventImpl(Component.java:4881)
	at java.awt.Container.dispatchEventImpl(Container.java:2292)
	at java.awt.Component.dispatchEvent(Component.java:4703)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4898)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4533)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4462)
	at java.awt.Container.dispatchEventImpl(Container.java:2278)
	at java.awt.Window.dispatchEventImpl(Window.java:2739)
	at java.awt.Component.dispatchEvent(Component.java:4703)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:746)
	at java.awt.EventQueue.access$400(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:697)
	at java.awt.EventQueue$3.run(EventQueue.java:691)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:719)
	at java.awt.EventQueue$4.run(EventQueue.java:717)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:716)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: me.legrange.swap.SerialException: Malformed integer response '' to ATCH? comamnd
	at me.legrange.swap.SerialModem.readATasInt(SerialModem.java:130)
	at me.legrange.swap.SerialModem.getSetup(SerialModem.java:76)
	at me.legrange.panstamp.Network.getSetup(Network.java:457)
	... 90 more
```

**Comment by GideonLeGrange** [comment]
```
me.legrange.swap.DecodingException: Invalid numeric data: '3E2F0004000700040B05OK-Command mode' (4f 4b)
	at me.legrange.swap.SerialMessage.<init>(SerialMessage.java:105)
	at me.legrange.swap.SerialModem$Reader.run(SerialModem.java:268)
Caused by: java.lang.NumberFormatException: For input string: "OK"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at me.legrange.swap.SerialMessage.<init>(SerialMessage.java:103)
	... 1 more

```

**Comment by GideonLeGrange** [comment]
Error connecting to serial network on /dev/tty.usbserial-A800HNMV: Malformed integer response '(2C30)0003' (28324333302930303033) to ATCH? comamnd

**Comment by GideonLeGrange** [comment]
Suggested fixes:
* Swallow empty lines in ```SerialModem.Reader.run()```  to guard against ```Malformed integer response '' to ATCH? comamnd``` - simple
* Try to detect 'missing' \r breaks ```SerialModem.Reader.run()``` by selecting different parsing when in DATA mode vs COMMAND mode to help with ```3E2F0004000700040B05OK-Command mode```- more difficult
* No idea how to fix ```(2C30)0003' (28324333302930303033) to ATCH? ```, can't see how this is not the modem sketch or serial driver
